packages/clarity-js/src/layout/dom.ts

```diff
@@ -115,10 +116,12 @@ export function add(node: Node, parent: Node, data: NodeInfo, source: Source): v
         selector: null,
         hash: null,
         region: regionId,
-        metadata: { active: true, suspend: false, privacy, position: null, size: null }
+        metadata: { active: true, suspend: false, privacy, position: null, size: null },
+        fragment: fragmentId,
     };
 
     updateSelector(values[id]);
+    if (values[id].fragment && updatedFragments.indexOf(values[id].fragment) === -1) { updatedFragments.push(values[id].fragment); }
```

Comment:
>values[id].fragment [](http://example.com/codeflow?start=8&length=19)

Minor: Just call this fragmentId as it's clearer and simpler to understand. It's the same value and unchanged from what your used above.